### DISCUSSION TITLE
Very bad performance. Help :(

### DISCUSSION THREAD

**Discussion by BerkeAras** [ionic-team/capacitor#3476] [Q&A] [answered]
Hello,

unfortunately my app has a very bad performance both on iOS and Android. The web version is very fast and powerful. I **do not use any frameworks**. What kind of tips could you give me? Should I include ionic locally or by the CDN?

**Comment by mhartington** [comment]
Hey there! There's not a lot of information to go off of from this. Can you please provide more details? "Bad performance" is pretty vague.

**Reply by BerkeAras** [reply]
As you can see in this video, it is very laggy. Sometimes it hangs for 2-3 seconds, or the ripple effect of the buttons are very stuttery. See here:
[https://www.youtube.com/watch?v=z5eqC4249QM](https://www.youtube.com/watch?v=z5eqC4249QM)

**Reply by mhartington** [reply]
A video still doesn't really tell us enough. 

- Do you have a sample project? 
- What framework are you using?
- Have you looked at the devtools and checked the performance there?

We need more information from you. 
Though this doesn't seem to be a capacitor specific issue, as it's most performance of your web app.

**Reply by BerkeAras** [reply]
As I already mentioned before, I do not use any frameworks, so I use @ionic/core. I've checked the devtools, but it is very very fast on web. Actually, I cannot give a sample project, I'm sorry :(

**Reply by mhartington** [reply]
>I've checked the devtools, but it is very very fast on web. 

Check the devtools when the app is running on the device.
Since you're seeing issue with android/chrome, open chrome and go to `chrome://inspect` and select the device/target.

**Reply by BerkeAras** [reply]
@mhartington 

![image](https://user-images.githubusercontent.com/36629890/91457092-49d14500-e884-11ea-8a84-5721d84a84fd.png)

What do you need more? That is just the app loading (which is fast).

On Android everything is very slow.

**Reply by mhartington** [reply]
Can't say for sure. This is very open-ended, and without having the project to inspect and look at myself, I can't say for sure what the issue is. Though, capacitor is probably not the culprit. I would suspect you need to look over your app while running on your android device


I just published this video to show how to connect when it is deployed to a real device. 

https://www.youtube.com/watch?v=7XgrrO1h7Wo

**Reply by BerkeAras** [reply]
@mhartington Thanks for creating the video. I've sent you an invitation for my private repository. Is it possible for you to look over it? Thanks 😊

**Comment by mhartington** [comment, marked as answer]
Ok, without going into tooo much detail, as there is a lot... here's what I got.


- Don't preload all the things

Your app is currently preloading just about everything when it doesn't need to. This could cause delays in start times and could be a bottle neck on lower end devices.

-  Don't use FastClick

You're including fast click in your app, when you do not need to. Modern browser do not have a click delay, making fast-click not needed. In your video gestures and other delays in clicks/drags could just be related to fast click. Remove it.

- Component composition

You're mixing jquery into web component classes, which you shouldn't do IMO. Honestly, you probably don't need jquery as most modern browser run es6, which is more than capable these days.

If you're seeing good performance on the desktop, it's not going to guarantee that the app will be performant on a real device. I honestly would suggest reworking your app to either use a framework like react or angular which provides best practices out of the box.


Performance issues are related to your code, not capacitor.

**Reply by BerkeAras** [reply]
I tried without preloading and the performance is slower. Thanks for your help.

**Reply by mhartington** [reply]
My twitter is mine. I simply stated something that could have been about anything. Either way, I deleted it.

**Comment by stefanhuber** [comment]
You have mentioned the performance problems appear only on Android and not on iOS? I am just wondering how many different Android devices you have used to test the performance and what Android versions?
